src/main/java/com/minecolonies/coremod/blocks/BlockTimberFrame.java

```diff
@@ -13,51 +25,112 @@
 public class BlockTimberFrame extends AbstractBlockMinecolonies<BlockTimberFrame>
 {
 
-    /**
-     * The hardness this block has.
-     */
-    private static final float BLOCK_HARDNESS = 3F;
+    public static final PropertyEnum<TimberFrameType> TYPE       = PropertyEnum.create("type", TimberFrameType.class);
 
     /**
      * This blocks name.
      */
-    private static final String BLOCK_NAME = "blockTimberFrame";
-
+    public static final String                      BLOCK_NAME     = "blockTimberFrame";
+    /**
+     * The hardness this block has.
+     */
+    private static final float                      BLOCK_HARDNESS = 3F;
     /**
      * The resistance this block has.
      */
-    private static final float RESISTANCE = 1F;
-
+    private static final float                      RESISTANCE     = 1F;
     /**
      * Constructor for the TimberFrame
      */
-    public BlockTimberFrame()
+    BlockTimberFrame(final String name)
     {
         super(Material.WOOD);
-        initBlock();
+        this.setDefaultState(this.blockState.getBaseState().withProperty(TYPE, TimberFrameType.PLAIN));
+        initBlock(name);
     }
-
     /**
      * initialize the block
      * sets the creative tab, as well as the resistance and the hardness.
      */
-    private void initBlock()
+    private void initBlock(final String name)
     {
-        setRegistryName(BLOCK_NAME);
-        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(), BLOCK_NAME));
+        setRegistryName(name);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(Locale.US), name));
         setCreativeTab(ModCreativeTabs.MINECOLONIES);
         setHardness(BLOCK_HARDNESS);
         setResistance(RESISTANCE);
     }
 
     /**
-     * Used to determine ambient occlusion and culling when rebuilding chunks for render.
+     * Registery block at gameregistry.
      *
-     * @return true
+     * @param registry the registry to use.
      */
+    @Override
+    public void registerItemBlock(final IForgeRegistry<Item> registry)
+    {
+        registry.register((new ItemColored(this, true)).setRegistryName(this.getRegistryName()));
+    }
+
+    /**
+     * Get the MapColor for this Block and the given BlockState
+     */
+    @NotNull
+    @Deprecated
+    @Override
+    public MapColor getMapColor(@NotNull final IBlockState state)
+    {
+        return state.getValue(TYPE).getMapColor();
+    }
+
+    /**
+     * Convert the given metadata into a BlockState for this Block
+     */
+    @NotNull
+    @Deprecated
+    @Override
+    public IBlockState getStateFromMeta(final int meta)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @deprecated Javadoc tag. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)